relay-server/src/actors/global_config.rs

```diff
@@ -253,25 +283,15 @@ impl Service for GlobalConfigService {
         tokio::spawn(async move {
             let mut shutdown_handle = Controller::shutdown_handle();
 
-            relay_log::info!("global config service starting");
-
-            if self.config.has_credentials() {
```

Comment:
I'd suggest we deal with the relay mode here, and maybe simplify a bit:

- If the mode is managed, run the code below. This requests from upstream and ignores the file.
- If the mode is proxy or static, we send one update to the watch channel with the config we read from the file and never request from upstream (see `handle_shutdown` function above).

Reply:
hm, so how about the case of no credentials for managed mode. And is there always no global config from upstream with static/proxy? should we abort relay if the file is missing?

Reply:
The number of combinations of these 3 is high and I think having a `match` statement cover all of them is hard to understand. What do you think about something like the following?

```rust
match mode {
    Proxy | Static | Capture => {
        if let Some(from_file) = config.global_config() {
            watch.send(from_file);
        }
        // else, the default is already there.
        // Alternatively, we can use `unwrap_or_default` and always send it to the channel.
    }
    Managed => {
        if config.has_credentials() {
            update_global_config();
        } else {
            relay_log();
        }
    }
}
```

Note that the sleep handle is initialized to `idle` (indefinitely pending) and it's not scheduled until a response from upstream is received. This means that not making an initial fetch from upstream results in never doing another fetch.

Reply:
ok it seems good actually, I changed it to that. 

> Note that the sleep handle is initialized to idle (indefinitely pending) and it's not scheduled until a response from upstream is received. This means that not making an initial fetch from upstream results in never doing another fetch. 
  
I did make an intial fetch when i call the `request_global_config` method